utils/zerovec/src/zerovec/mod.rs

```diff
@@ -81,6 +81,7 @@ use core::ops::Deref;
 ///
 /// See [the design doc](https://github.com/unicode-org/icu4x/blob/main/utils/zerovec/design_doc.md) for more details.
 #[derive(Clone)]
+#[non_exhaustive]
```

Comment:
Thought: At this point I don't know what other variants we would add, but we can always relax the `non_exhaustive` restriction in the future.

Reply:
yeah